test/Concurrency/Runtime/async_task_locals_in_task_group_may_need_to_copy.swift

```diff
@@ -0,0 +1,104 @@
+// RUN: %target-run-simple-swift( -plugin-path %swift-plugin-dir -Xfrontend -strict-concurrency=complete -Xfrontend -disable-availability-checking -parse-as-library %import-libdispatch) | %FileCheck %s --dump-input=always
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: libdispatch
+// REQUIRES: concurrency_runtime
+// UNSUPPORTED: back_deployment_runtime
+
+enum TL {
+  @TaskLocal
+  static var one: Int = 1
+  @TaskLocal
+  static var two: Int = 2
+  @TaskLocal
+  static var three: Int = 3
+}
+
+// ==== ------------------------------------------------------------------------
+
+func test() async {
```

Comment:
Can we add a test with multiple nested task groups inside child tasks? Just to make sure we don't copy above a `withTaskGroup` scope

Reply:
That’sa good test to add thx. We will copy those too, too defensively tbh. Preventing that is rough as we run out of space for some flags detecting this… maybe optimizing that copy away is a separate future follow uo